crates/revm/revm-inspectors/src/tracing/builder/parity.rs

```diff
@@ -637,3 +562,76 @@ where
 
     Ok(())
 }
+
+/// Returns the number of items pushed on the stack by a given opcode.
+/// This used to determine how many stack etries to put in the `push` element
+/// in a parity vmTrace.
+/// The value is obvious for most opcodes, but SWAP* and DUP* are a bit weird,
+/// and we handle those as they are handled in parity vmtraces.
+/// For reference: <https://github.com/ledgerwatch/erigon/blob/9b74cf0384385817459f88250d1d9c459a18eab1/turbo/jsonrpc/trace_adhoc.go#L451>
+pub(crate) fn stack_push_count(step_op: OpCode) -> usize {
+    let step_op = step_op.get();
+    if (opcode::PUSH0..=opcode::PUSH32).contains(&step_op) {
```

Comment:
I believe this can be a single match expression just like the Go switch reference

Reply:
> I believe this can be a single match expression just like the Go switch reference

Good call, just pushed that change.